Update item-components.md

StealthyExpertX opened the pull request:
Updated the documentation for the Enchantable item component to include detailed notes of the "value" key, which was previously lacking.

StealthyExpertX committed:
Update item-components.md

Updated the documentation for the Enchantable item component to include detailed notes of the "value" key, which was previously lacking.

QuazChick commented on docs/items/item-components.md:
```diff
@@ -199,6 +199,24 @@ Note: The "all" enchantable slot allows you to apply any enchantment that you wa
+| Netherite | 15                   | 15                        |
+| Other     | 1                    | 1                         |
+
+For an in-depth exploration of enchantability and its impact on the game, refer to the [Enchanting Mechanics on the Unofficial Minecraft Wiki](https://minecraft.fandom.com/wiki/Enchanting_mechanics#Enchantability).
```
Update to new Minecraft Wiki URL: https://minecraft.wiki/w/Enchanting_mechanics#Enchantability

StealthyExpertX committed:
Update item-components.md

Updated URL for wiki link for enchant info.

https://minecraft.wiki/w/Enchanting_mechanics#Enchantability
